test/Autocomplete.spec.js

```diff
@@ -1,176 +1,59 @@
-  });
+  let wrapper;
 
-  test('generates correct ID for input and label', () => {
-    expect(wrapper.find('.autocomplete').props()).toHaveProperty(
-      'id',
-      componentId
-    );
-    expect(wrapper.find('label').props()).toHaveProperty(
-      'htmlFor',
-      componentId
+  test('renders', () => {
+    wrapper = shallow(
+      <Autocomplete title="Test Title" id={componentId} options={{ data }} />
     );
+    expect(wrapper).toMatchSnapshot();
   });
 
-  describe('on input change', () => {
-    beforeAll(() => {
-      wrapper
-        .find('.autocomplete')
-        .simulate('change', { target: { value: typedKey } });
-    });
-
-    test('renders a dropdown', () => {
-      expect(wrapper.find('.autocomplete-content')).toHaveLength(1);
-    });
-
-    test("highlight's results", () => {
-      expect(wrapper.find('.highlight').length).toEqual(2);
-      expect(
-        wrapper
-          .find('.highlight')
-          .first()
-          .text()
-      ).toEqual(typedKey);
-    });
-  });
-
-  describe('on dropdown first item select', () => {
-    const expectedValue = 'Apple';
-
-    beforeAll(() => {
-      wrapper
-        .find('.autocomplete')
-        .simulate('change', { target: { value: typedKey } });
-      wrapper
-        .find('ul li')
-        .first()
-        .simulate('click');
-    });
-
-    test('updates the state with the new value', () => {
-      expect(wrapper.state('value')).toEqual(expectedValue);
-    });
-
-    test('adds clicked value to input', () => {
-      expect(wrapper.find('.autocomplete').prop('value')).toEqual(
-        expectedValue
-      );
-    });
-
-    test('hides other matches', () => {
-      expect(wrapper.find('.highlight').length).toEqual(0);
-    });
+  test('handles layout classes', () => {
+    wrapper = shallow(<Autocomplete s={4} m={6} />);
+    expect(wrapper.hasClass('input-field')).toBeTruthy();
+    expect(wrapper.hasClass('col')).toBeTruthy();
+    expect(wrapper.hasClass('s4')).toBeTruthy();
+    expect(wrapper.hasClass('m6')).toBeTruthy();
   });
 
-  describe('on dropdown second item select', () => {
-    const expectedValue = 'Apple Inc';
-
-    beforeAll(() => {
-      wrapper
-        .find('.autocomplete')
-        .simulate('change', { target: { value: typedKey } });
-      wrapper
-        .find('ul li')
-        .last()
-        .simulate('click');
-    });
-
-    test('updates the state with the new value', () => {
-      expect(wrapper.state('value')).toEqual(expectedValue);
-    });
-
-    test('adds clicked value to input', () => {
-      expect(wrapper.find('.autocomplete').prop('value')).toEqual(
-        expectedValue
-      );
-    });
-
-    test('hides other matches', () => {
-      expect(wrapper.find('.highlight').length).toEqual(0);
-    });
+  test('renders placeholder', () => {
+    wrapper = shallow(<Autocomplete placeholder="Name" />);
+    expect(wrapper.find('input').prop('placeholder')).toEqual('Name');
```

Comment:
I'm never really sure this kind of assertion is useful, since we just passed that prop, it will for sure have that prop. What happens to the DOM in this case? Can we test that?

Reply:
So, should I delete this test?

Reply:
Nope, it's fine :)